node/core/pvf/src/execute/worker.rs

```diff
@@ -246,7 +246,12 @@ async fn validate_using_artifact(
 
 	let validation_started_at = Instant::now();
 	let descriptor_bytes =
-		match crate::executor_intf::execute(compiled_artifact, params, spawner.clone()) {
+		match unsafe {
+			// SAFETY: this should be safe since the compiled artifact passed here comes from the
+			//         file created by the prepare workers. These files are obtained by calling
+			//         [`executor_intf::prepare`].
+			crate::executor_intf::execute(compiled_artifact, params, spawner.clone())
```

Comment:
I am not sure if this is enough TBH. The data that comes here is from a file. The file can in theory be compromised or corrupted (e.g. due to hardware failure). 

However, there is also little sense in extending the `unsafe` blocks further. It's awkward and will probably lead marking all the worker_common machinery. Yet, it won't provide any additional benefit - we ultimately will walk into the cross process boundary. Ultimately, it's up to validation host to decide what file will be passed here.

I think we do not take into account the compromisation concern. We mostly assume that if the node is compromised the attacker can do way worse things anyway. The corruption is a bit more severe. However, I am not sure if it's likely: after all, to my knowledge, ELFs do not have checksums.

Reply:
If you want to be 100% sure, you could hash them. However, I'm not sure that is really required.